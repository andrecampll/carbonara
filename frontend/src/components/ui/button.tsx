import { Button as BaseButton, ButtonProps } from '@mui/base/Button'
import clsx from 'clsx'
import * as React from 'react'

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const { className, ...other } = props
    return (
      <BaseButton
        ref={ref}
        className={clsx(
          'ui-disabled:text-slate-700 ui-disabled:dark:text-slate-200 ui-disabled:bg-slate-200 ui-disabled:dark:bg-slate-700 ui-disabled:cursor-default ui-disabled:shadow-none ui-disabled:dark:shadow-none ui-disabled:hover:bg-slate-200 ui-disabled:hover:dark:bg-slate-700 ui-disabled:border-none hover:bg-secondary-600 active:bg-secondary-700 cursor-pointer rounded-lg border border-solid border-secondary-500 bg-secondary-500 px-4 py-2 font-sans text-sm font-semibold leading-normal text-white transition focus-visible:outline-none active:scale-[0.99] active:shadow-none',
          className,
        )}
        {...other}
      />
    )
  },
)

Button.displayName = 'Button'
