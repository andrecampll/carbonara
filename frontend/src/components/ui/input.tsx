import { TextField, TextFieldProps } from '@mui/material'
import { forwardRef } from 'react'

export type InputProps = TextFieldProps

export const Input = forwardRef<HTMLDivElement, InputProps>(
  ({ ...props }, ref) => {
    return (
      <TextField
        fullWidth
        color="secondary"
        {...props}
        error={!!props.helperText}
        ref={ref}
      />
    )
  },
)

Input.displayName = 'Input'
