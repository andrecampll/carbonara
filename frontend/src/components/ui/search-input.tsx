import { Autocomplete } from '@mui/material'
import { forwardRef } from 'react'

import { Input, InputProps } from './input'

type Props = InputProps & {
  onChangeValue: (value: string) => void
}

export const SearchInput = forwardRef<HTMLDivElement, Props>(
  ({ label, helperText, onChangeValue }, ref) => {
    const dietaryChoices = [
      {
        value: 'Vegan',
        label: 'Vegan',
      },
      {
        value: 'Vegetarian',
        label: 'Vegetarian',
      },
      {
        value: 'Pescatarian',
        label: 'Pescatarian',
      },
      {
        value: 'MeatEater',
        label: 'Meat Eater',
      },
    ]

    return (
      <div>
        <Autocomplete
          options={dietaryChoices}
          renderInput={(params) => (
            <Input {...params} label={label} helperText={helperText} />
          )}
          onChange={(_, option) => option && onChangeValue(option.value)}
          ref={ref}
          isOptionEqualToValue={(option, value) => option.value === value.value}
        />
      </div>
    )
  },
)

SearchInput.displayName = 'SearchInput'
