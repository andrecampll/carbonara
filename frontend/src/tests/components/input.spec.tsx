import { render, screen } from '@testing-library/react'
import { describe, expect, test } from 'vitest'

import { Input } from '../../components/ui/input'

describe('Input', () => {
  test('it should be able to render Input component', () => {
    const { container } = render(<Input />)

    expect(screen.getByRole('textbox')).toBeDefined()

    expect(container.firstChild).toMatchSnapshot()
  })
})
