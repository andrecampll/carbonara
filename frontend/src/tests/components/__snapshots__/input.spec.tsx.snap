// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > it should be able to render Input component 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      id=":r0:"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-ihdtdm"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
